server/historian/packages/historian-base/src/runnerFactory.ts

```diff
@@ -195,18 +195,17 @@ export class HistorianResourcesFactory implements core.IResourcesFactory<Histori
 			? customizations?.storageNameRetriever ?? new services.StorageNameRetriever()
 			: undefined;
 
+		const port = normalizePort(process.env.PORT || "3000");
 		const tenantManager: core.ITenantManager = new services.TenantManager(
 			riddlerEndpoint,
-			undefined /* internalHistorianUrl */,
+			`http://localhost:${port}` /* internalHistorianUrl (self) */,
```

Comment:
Would if `TenantManager` have some special logic for `internalHistorianUrl` when it is `undefined`? Is it safe to set to self?

Reply:
The reasons for this change are precisely that `TenantManager` _does not_ have logic to handle `undefined` [where it uses `internalHistorianUrl`](https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/services/src/tenant.ts#L141). What would happen is Historian would send a request to `undefined/repos/<tenantId>`.

Regardless, `internalHistorianUrl` from Historian should be its own local API port, so this change is logical and correct.

Lastly, the reason this was historically set to `undefined` is that Historian itself doesn't use the `getTenantGitManager` function of TenantManager, so it doesn't hit the part of code that would call the undefined URL. The use of `undefined` here was a bit of a lazy "well this param won't be hit so it doesn't matter."

Reply:
_If_ this URL was hit, it would probably create a circular API call which would infinitely happen, though, so I agree that is bad. I will change the URL to something explicitly invalid.